core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java

```diff
@@ -1741,8 +1744,17 @@ public String getSubclassForDiscriminatorValue(Object value) {
 		return spaces;
 	}
 
-	public TupleContext getTupleContext() {
-		return tupleContext;
+	/**
+	 * Returns the {@link TupleContext}.
+	 *
+	 * @param session the current session, if null the {@link TupleContext#getTransactionContext()} will be null.
+	 * @return the tupleContext for the session
+	 */
+	public TupleContext getTupleContext( SessionImplementor session ) {
```

Comment:
I need to add the transaction id to the TupleContext, I get the transaction id from the session.

I can pass the TransactionContext, instead of the session I guess, but I don't think the amount of time I need to call TransactionContextHelper will change.

I guess I could add an interface to identify which dialects need a transaction context, would that be better?